packages/lib/bookingSuccessRedirect.ts

```diff
@@ -8,23 +8,34 @@ import { useCompatSearchParams } from "@calcom/lib/hooks/useCompatSearchParams";
 import { navigateInTopWindow } from "@calcom/lib/navigateInTopWindow";
 
 function getNewSearchParams(args: {
-  query: Record<string, string | null | undefined | boolean>;
+  query: Record<
+    string,
+    string | null | undefined | boolean | { name?: string } | Array<{ name: string; email?: string }>
+  >;
   searchParams?: URLSearchParams;
 }) {
   const { query, searchParams } = args;
   const newSearchParams = new URLSearchParams(searchParams);
+
   Object.entries(query).forEach(([key, value]) => {
-    if (value === null || value === undefined) {
-      return;
+    if (value === null || value === undefined) return;
+
+    if (Array.isArray(value) && key === "attendees" && value.length > 0) {
+      const { name } = value[0];
+      if (name) newSearchParams.append("attendeeName", name);
+    } else if (typeof value === "object" && value !== null && "name" in value && key === "user") {
+      newSearchParams.append("hostName", value.name || "");
+    } else {
+      newSearchParams.append(key, String(value));
```

Comment:
In case of team event types where there are more than one host. other hosts are stored in attendees table